src/drive/mobile/modules/mediaBackup/duck/index.js

```diff
@@ -34,9 +37,95 @@ const currentMediaUpload = (media, uploadCounter, totalUpload) => ({
   }
 })
 
-const getDirID = async path => {
-  const { _id } = await cozy.client.files.createDirectoryByPath(path, false)
-  return _id
+const getFoldersWithReference = async reference => {
+  const referencedFolders = await cozy.client.fetchJSON(
+    'POST',
+    '/files/_find',
+    {
+      selector: {
+        referenced_by: {
+          $elemMatch: {
+            type: reference
+          }
+        }
+      },
+      sort: [{ created_at: 'desc' }]
+    }
+  )
+
+  return referencedFolders.filter(
+    folder => /^\/\.cozy_trash/.test(folder.attributes.path) === false
+  )
+}
+
+const addFolderReference = (folderId, reference) =>
+  cozy.client.fetchJSON(
+    'POST',
+    `/files/${folderId}/relationships/referenced_by`,
+    {
+      data: [
+        {
+          type: reference,
+          id: reference
+        }
+      ]
+    }
+  )
+
+const ensureFolderWithReferenceExists = async (path, reference) => {
+  const referencedFolders = await getFoldersWithReference(reference)
+
+  if (referencedFolders.length) {
+    return referencedFolders[0]
+  } else {
+    const dir = await cozy.client.files.createDirectoryByPath(path, false)
+    await addFolderReference(dir._id, reference)
+    return dir
+  }
+}
+
+const ensureHasCreatedAtIndex = () =>
+  cozy.client.fetchJSON('POST', '/data/io.cozy.files/_index', {
+    index: {
+      fields: ['created_at']
+    }
+  })
+
+const getUploadDir = async () => {
+  await ensureHasCreatedAtIndex()
```

Comment:
Why do we need to check if the `created_at` index is created ?

Reply:
Because in `getFoldersWithReference`, we sort the files by `created_at`, and if there is no index beforehand Couch can't resolve the request.